engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -62,13 +69,23 @@ class EngineJobExecutionActor(replyTo: ActorRef,
   // When Pending, the FSM always has NoData
   when(Pending) {
     case Event(Execute, NoData) =>
+      requestExecutionToken()
+      goto(RequestingExecutionToken)
+  }
+
+  when(RequestingExecutionToken) {
+    case Event(JobExecutionTokenDispensed(jobExecutionToken), NoData) =>
+      executionToken = Option(jobExecutionToken)
       if (restarting) {
         val jobStoreKey = jobDescriptorKey.toJobStoreKey(workflowId)
         jobStoreActor ! QueryJobCompletion(jobStoreKey, jobDescriptorKey.call.task.outputs)
         goto(CheckingJobStore)
       } else {
         prepareJob()
       }
+    case Event(JobExecutionTokenDenied(positionInQueue), NoData) =>
+      log.debug("Token denied so cannot start yet. Currently position {} in the queue", positionInQueue)
+      stay()
```

Comment:
I originally had it doing this, but realised we could end up with 10s of thousands of actors requesting tokens every second for no purposeful reason since they know that they're already queued up if they get the "Denied" response (and forcing the dispenser to traverse its queue every time to find the index seems wasteful!)